src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.cpp

```diff
@@ -0,0 +1,197 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *
+ *    Copyright (c) 2020 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/***************************************************************************/
+/**
+ * @file
+ * @brief Routines for the Occupancy plugin, which
+ *implements the Occupancy server cluster.
+ *******************************************************************************
+ ******************************************************************************/
+#include "occupancy-sensor-server.h"
+
+#include <app/util/af.h>
+
+#include <app/util/af-event.h>
+#include <app/util/attribute-storage.h>
+
+#include "gen/att-storage.h"
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+#include "gen/command-id.h"
+#include "gen/enums.h"
+
+#include "occupancy-hal.h"
+
+#ifdef EMBER_AF_PLUGIN_REPORTING
+#include <app/reporting/reporting.h>
+#endif
+
+using namespace chip;
+
+//------------------------------------------------------------------------------
+// Plugin private function prototypes
+static void checkForReportingConfig(void);
+
+//------------------------------------------------------------------------------
+// Global variables
+/*
+// The default configuration to be used if no reporting has been set up
```

Comment:
Why is all the reporting here getting commented out?

Reply:
did not have the means to test reporting and don't know whether the code from SiLabs is still required with CHIP.

Reply:
Reporting generally works with CHIP afaik, and uses the silabs code at the moment.

Reply:
checked reporting of occupancy and it works. removed the commented code.